GVFS/GVFS/CommandLine/StatisticsVerb.cs

```diff
@@ -0,0 +1,209 @@
+﻿using CommandLine;
+using GVFS.Common;
+using GVFS.Common.Database;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.NamedPipes;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.CommandLine
+{
+    [Verb(StatisticsVerb.StatisticsVerbName, HelpText = "Get statistics for the health state of the repository")]
+    public class StatisticsVerb : GVFSVerb.ForExistingEnlistment
+    {
+        private const string StatisticsVerbName = "statistics";
+
+        private const int DirectoryDisplayCount = 5;
+
+        [Option(
+            'd',
+            "directory",
+            Required = false,
+            HelpText = "Run the statistics tool on a specific directory")]
+        public string Directory { get; set; }
+
+        protected override string VerbName
+        {
+            get { return StatisticsVerbName; }
+        }
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            // The path to the root of git's tree
+            string sourceRoot = enlistment.DotGitRoot.Substring(0, enlistment.DotGitRoot.Length - 4);
+
+            if (this.Directory == null)
+            {
+                this.Directory = string.Empty;
+            }
+            else if (this.Directory.Equals("."))
+            {
+                this.Directory = Environment.CurrentDirectory.Substring(sourceRoot.Length);
+            }
+
+            // Get all of the data needed to calculate statistics then pass them into GVFSEnlistmentStatistics
+            List<string> modifiedPathsFileList = new List<string>();
+            List<string> modifiedPathsFolderList = new List<string>();
+            List<string> placeholderFilePathList = new List<string>();
+            List<string> placeholderFolderPathList = new List<string>();
+
+            this.GetPlaceholdersFromDatabase(enlistment, out List<IPlaceholderData> filePlaceholders, out List<IPlaceholderData> folderPlaceholders);
+            foreach (IPlaceholderData placeholderData in filePlaceholders)
+            {
+                placeholderFilePathList.Add(placeholderData.Path);
+            }
+
+            foreach (IPlaceholderData placeholderData in folderPlaceholders)
+            {
+                placeholderFolderPathList.Add(placeholderData.Path);
+            }
+
+            foreach (string path in this.GetModifiedPathsFromPipe(enlistment))
+            {
+                if (path.Last() == '/')
+                {
+                    modifiedPathsFolderList.Add(path.Take(path.Length - 1).ToString());

```

Comment:
path.TrimEnd('/'), instead of doing math?

Reply:
Didn't realize there was a function for that, stuck it in there 👍